components/cdi/weld/src/main/java/com/kumuluz/ee/configuration/cdi/interceptors/ConfigBundleInterceptor.java

```diff
@@ -62,96 +66,186 @@ public Object loadConfiguration(InvocationContext ic) throws Exception {
+                // get field annotation - @ConfigValue
+                Field field = targetClass.getDeclaredField(setterToField(method.getName()));
+                ConfigValue fieldAnnotation = null;
+                if (field != null) {
+                    fieldAnnotation = field.getAnnotation(ConfigValue.class);
+                }
 
-                    if (value.isPresent()) {
-                        m.invoke(target, value.get());
+                // watch nested class or list if all fields in the bean are annotated with watch or if a field is
+                // annnotated with watch
+                boolean watchNestedClass = watchAllFields;
+                if (watchNestedClass == false) {
+                    if (fieldAnnotation != null) {
+                        watchNestedClass = fieldAnnotation.watch();
                     }
+                }
 
-                    deployWatcher(target, targetClass, m, getKeyName(targetClass, m.getName()));
-
-                } else if (m.getParameters()[0].getType().equals(Float.class)) {
+                // process primitives
+                if (Arrays.asList(primitives).contains(parameterType)) {
 
-                    Optional<Float> value = configurationUtil.getFloat(getKeyName(targetClass, m.getName()));
+                    Optional<String> value = getValueOfPrimitive(parameterType, getKeyName(targetClass, method
+                            .getName(), keyPrefix));
 
                     if (value.isPresent()) {
-                        m.invoke(target, value.get());
+                        method.invoke(target, value.get());
                     }
 
-                    deployWatcher(target, targetClass, m, getKeyName(targetClass, m.getName()));
-
-                } else if (m.getParameters()[0].getType().equals(Double.class)) {
+                    if (watchAllFields || (fieldAnnotation != null && fieldAnnotation.watch())) {
+                        deployWatcher(target, method, getKeyName(targetClass, method.getName(), keyPrefix));
+                    }
 
-                    Optional<Double> value = configurationUtil.getDouble(getKeyName(targetClass, m.getName()));
+                    // process nested objeccts
+                } else if (!parameterType.isArray()) {
 
-                    if (value.isPresent()) {
-                        m.invoke(target, value.get());
-                    }
+                    processedClassRelations.put(targetClass, parameterType);
 
-                    deployWatcher(target, targetClass, m, getKeyName(targetClass, m.getName()));
+                    Object nestedTarget = processNestedObject(targetClass, method, parameterType, keyPrefix,
+                            processedClassRelations, -1, watchNestedClass);
 
-                } else if (m.getParameters()[0].getType().equals(Integer.class)) {
+                    // invoke setter method with initialised instance
+                    method.invoke(target, nestedTarget);
```

Comment:
Perhaps have the same functionality as the primitives, as in don't set the variable if it doesn't exist or in this case can not be constructed. For instance if no keys for the nested class exist leave it as is, because the user might have a default one set in code.

Reply:
Fixed.